srcdeps-maven-local-repository/src/test/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManagerTest.java

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@RunWith(MavenJUnitTestRunner.class)
+@MavenVersions({ "3.3.1" })
+public class SrcdepsLocalRepositoryManagerTest {
+    private static final Logger log = LoggerFactory.getLogger(SrcdepsLocalRepositoryManagerTest.class);
+
+    private static final Path mvnLocalRepo;
+    private static final String mrmSettingsXmlPath = System.getProperty("mrm.settings.xml");
+    private static final Path basedir = Paths.get(System.getProperty("basedir", new File("").getAbsolutePath()));
+    private static final Path projectsSrcDirectory;
+    private static final Path projectsTargetDirectory;
+
+    static {
+        projectsSrcDirectory = basedir.resolve("src/projects");
+        projectsTargetDirectory = basedir.resolve("target/projects");
+        mvnLocalRepo = basedir.resolve("target/mvn-local-repo");
+    }
+
+    public final MavenRuntime verifier;
+
+    @Rule
+    public final TestResources resources = new TestResources();
+    protected String currentTestName;
+
+    @Rule
+    public TestRule watcher = new TestWatcher() {
+
+        @Override
+        protected void starting(Description description) {
+
+            final String rawMethodName = description.getMethodName();
+            final int bracePos = rawMethodName.indexOf('[');
+            SrcdepsLocalRepositoryManagerTest.this.currentTestName = bracePos >= 0
+                    ? rawMethodName.substring(0, bracePos) : rawMethodName;
+        }
+
+    };
+
+    @BeforeClass
+    public static void beforeClass() {
+        Assert.assertTrue("[" + mrmSettingsXmlPath + "] should exist", Files.exists(Paths.get(mrmSettingsXmlPath)));
+    }
+
+    public SrcdepsLocalRepositoryManagerTest(MavenRuntimeBuilder runtimeBuilder) throws Exception {
+        this.verifier = runtimeBuilder.withExtension(new File("target/classes").getCanonicalFile()).build();
+    }
+
+    public MavenExecutionResult assertBuild(String artifactId, String srcVersion, String goal) throws Exception {
+        final String artifactDir = "org/l2x6/maven/srcdeps/itest/" + artifactId;
+        SrcdepsCoreUtils.deleteDirectory(mvnLocalRepo.resolve(artifactDir));
+
+        MavenExecution execution = verifier.forProject(resources.getBasedir(currentTestName)) //
+                //.withCliOption("-X") //
+                .withCliOptions("-Dmaven.repo.local=" + mvnLocalRepo.toAbsolutePath().toString())
+                .withCliOption("-s").withCliOption(mrmSettingsXmlPath)
+        ;
+        MavenExecutionResult result = execution.execute("clean", "compile");
+        result //
+                .assertErrorFreeLog() //
+                .assertLogText(
+                        "SrcdepsLocalRepositoryManager will decorate org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManagerFactory with priority 10.0") //
+        ;
+
+        final String artifactPrefix = artifactDir + "/" + srcVersion + "/" + artifactId + "-" + srcVersion;
```

Comment:
AFAIK, `java.io.File` is not picky about the delimiter char: I think that both `/` and `\` work on Windows.